pyretailscience/plots/bar.py

```diff
@@ -0,0 +1,190 @@
+        bar_width (float, optional): The width of the bars. Defaults to 0.35.
+        ax (Axes, optional): The Matplotlib Axes object to plot on. Defaults to None.
+        source_text (str, optional): Text to be displayed as a source at the bottom of the plot. Defaults to None.
+        move_legend_outside (bool, optional): Whether to move the legend outside the plot area. Defaults to False.
+        orientation (Literal["horizontal", "h", "vertical", "v"], optional): Orientation of the bars. Can be
+                                                                             "horizontal", "h", "vertical", or "v".
+                                                                             Defaults to "vertical".
+        sort_order (Literal["ascending", "descending"] | None, optional): Sorting order for the bars. Can be
+                                                                          "ascending" or "descending". Defaults to None.
+        data_label_format (Literal["absolute", "percentage"] | None, optional): Format for displaying data labels.
+                                                                                "absolute" shows raw values,
+                                                                                "percentage" shows percentage.
+                                                                                Defaults to None.
+        use_hatch (bool, optional): Whether to apply hatch patterns to the bars. Defaults to False.
+        **kwargs (dict[str, any]): Additional keyword arguments for the Pandas `plot` function.
+
+    Returns:
+        SubplotBase: The Matplotlib Axes object with the generated plot.
+    """
+    valid_orientations = ["horizontal", "h", "vertical", "v"]
+    if orientation not in valid_orientations:
+        error_msg = f"Invalid orientation: {orientation}. Expected one of {valid_orientations}"
+        raise ValueError(error_msg)
+
+    # Validate the sort_order value
+    valid_sort_orders = ["ascending", "descending", None]
+    if sort_order not in valid_sort_orders:
+        error_msg = f"Invalid sort_order: {sort_order}. Expected one of {valid_sort_orders}"
+        raise ValueError(error_msg)
+
+    # Validate the data_label_format value
+    valid_data_label_formats = ["absolute", "percentage", None]
+    if data_label_format not in valid_data_label_formats:
+        error_msg = f"Invalid data_label_format: {data_label_format}. Expected one of {valid_data_label_formats}"
+        raise ValueError(error_msg)
+
+    value_col = [value_col] if isinstance(value_col, str) else (["Value"] if value_col is None else value_col)
+
+    df = df.to_frame(name=value_col[0]) if isinstance(df, pd.Series) else df
+
+    df = df.sort_values(by=value_col[0], ascending=sort_order == "ascending") if sort_order is not None else df
+
+    cmap = get_multi_color_cmap() if len(value_col) > 1 else get_single_color_cmap()
```

Comment:
I think this should be 3. With the expectation of the histogram which has overlaps, I've been using the 3 shades of green for up to 3 colors and the multicolor pallette when more.
```suggestion
    cmap = get_multi_color_cmap() if len(value_col) > 3 else get_single_color_cmap()
```